src/main/java/com/google/step/servlets/UserServlet.java

```diff
@@ -70,10 +72,14 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
       return;
     }
 
+    String idString = request.getParameter("id");
+    if (idString == null) {
+      response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+      return;
+    }
     long id;
     User user;
     try {
-      String idString = request.getPathInfo().substring(1);
       id = Long.parseLong(idString);
       user = userManager.readUser(id);
     } catch (IndexOutOfBoundsException | IllegalArgumentException e) {
```

Comment:
Since you got rid of the request.getPathInfo().substring im not too sure if IndexOutOfBoundsException would still occur? and maybe u can add NumberFormatException for the parseLong(idString)